lib/matplotlib/widgets.py

```diff
@@ -905,27 +944,14 @@ def set_val(self, val):
         _api.check_shape((2,), val=val)
         val[0] = self._min_in_bounds(val[0])
         val[1] = self._max_in_bounds(val[1])
-        xy = self.poly.xy
+        self.poly.xy = self._fill_verts(self.poly.xy, val)
```

Comment:
```suggestion
        self._update_selection_poly(val[0], val[1])
```
Optional: It might be more readable if you unpack `vmin, vmax = val` and use these variables instead of `val[0]`, `val[1]`. Throughout the function.